src/main/java/io/confluent/connect/jdbc/source/JdbcSourceTask.java

```diff
@@ -149,21 +151,35 @@ public void start(Map<String, String> properties) {
                 timestampColumns
             );
           }
-          partition = Collections.singletonMap(
-              JdbcSourceConnectorConstants.TABLE_NAME_KEY,
-              tableOrQuery
-          );
+          partitionsForTable(partitions, tableOrQuery);
           break;
         case QUERY:
           partition = Collections.singletonMap(
               JdbcSourceConnectorConstants.QUERY_NAME_KEY,
               JdbcSourceConnectorConstants.QUERY_NAME_VALUE
           );
+          partitions.add(partition);
           break;
         default:
           throw new ConnectException("Unexpected query mode: " + queryMode);
       }
-      Map<String, Object> offset = offsets == null ? null : offsets.get(partition);
+
+      Map<String, Object> offset = null;
+      int offsetProtocolVersion = 0;
+      if (offsets != null) {
+        for (Map<String, String> toCheckPartition : partitions) {
+          offset = offsets.get(toCheckPartition);
+          if (offset != null) {
+            log.trace("Found non-null offset {} for partition {}", offsets, toCheckPartition);
+            offsetProtocolVersion = Integer.parseInt(
+                toCheckPartition.getOrDefault(
+                    JdbcSourceConnectorConstants.OFFSET_PROTOCOL_VERSION,
+                    JdbcSourceConnectorConstants.PROTOCOL_VERSION_ZERO
+                ));
```

Comment:
Would be good to have the TRACE log message _after_ getting the protocol version, and then including the protocol version in the log message.